server/src/appFactory.ts

```diff
@@ -23,10 +22,7 @@ function buildProjectURL(
   return `${scheme}://${host}/project/${projectId}`;
 }
 
-async function appFactory(
-  dynamoDb: AWS.DynamoDB.DocumentClient,
-  listoData: DirectoryData,
-) {
+async function appFactory(db: DatabaseAdapter, listoData: DirectoryData) {
```

Comment:
I don't think you need this adapter pattern, you should be able to describe `db` as being of a type that implements the `Database` interface. Thus, any class that implements this interface may be injected into the `appFactory` function.

The job of creating the right type of database could move to `app.ts` as part of the application boostrapping procedure.

Reply:
Agree and have removed the dbadapter :)